crates/shared/src/sources/balancer_v2/pool_fetching.rs

```diff
@@ -199,74 +194,63 @@ impl BalancerPoolFetching for BalancerPoolFetcher {
         token_pairs: HashSet<TokenPair>,
         at_block: Block,
     ) -> Result<FetchedBalancerPools> {
-        let pool_ids = self
-            .pool_registry
-            .pool_ids_for_token_pairs(&token_pairs)
-            .await;
-        let fetched_stable_pools = self
-            .stable_pool_reserve_cache
-            .fetch(pool_ids.clone(), at_block)
-            .await?;
-        let fetched_weighted_pools = self
-            .weighted_pool_reserve_cache
-            .fetch(pool_ids, at_block)
-            .await?;
-        // Return only those pools which are not paused.
-        Ok(FetchedBalancerPools {
-            stable_pools: filter_paused(fetched_stable_pools),
-            weighted_pools: filter_paused(fetched_weighted_pools),
-        })
-    }
-}
+        let pool_ids = self.fetcher.pool_ids_for_token_pairs(token_pairs).await;
```

Comment:
Do we need to have `pool_ids_for_token_pairs` and `pools_by_id` functions separated? I don't see any case when we call `pool_ids_for_token_pairs` without calling `pools_by_id` right after.

Reply:
Its because of caching details. Specifically, the pool cache operates on `Map<H256, Pool>` (since a single pool can be over multiple token pairs, and multiple pools can trade the same token pair).

Additionally, we use an LRU cache strategy where we eagerly update commonly used pools every time we observe a new block. This means that we need to have a `pools_by_id` to be able to only update those pools.